comms/dht/src/inbound/decryption.rs

```diff
@@ -248,7 +248,7 @@ where S: Service<DecryptedDhtMessage, Response = (), Error = PipelineError>
             Ok(origin_mac) => {
                 // If this fails, discard the message because we decrypted and deserialized the message with our shared
                 // ECDH secret but the message could not be authenticated
-                let msg_challenge = crypt::create_message_challenge(&message.dht_header, &message.body);
+                let msg_challenge = crypt::create_message_domain_separated_hash(&message.dht_header, &message.body);
```

Comment:
Recommend renaming to be consistent with the rest of the PR.

Reply:
Something like `binding_message_representation` might be nicely expressive if used consistently.

Reply:
I renamed these variables, as you suggested, as well as the terminology `OriginMac`/`origin_mac`